### PR TITLE
Update OTP versions

### PR DESCRIPTION
**Pull request by chrzaszcz**
Update versions of Erlang/OTP used in CircleCI and GH Actions.
Also: update the release branch in buttons.

See commits for details.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/214715) / elasticsearch_and_cassandra_mnesia / f76259639ff5b5c12238c34d9497e177d54e6503
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4260/214715/elasticsearch_and_cassandra_mnesia.26.2.3-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4260/214715/elasticsearch_and_cassandra_mnesia.26.2.3-amd64/big/ct_run.test%4092c41cb8bbde.2024-04-10_12.44.10/index.html)
**OK: 437** / Failed: 0 / **User-skipped: 43** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/214747) / elasticsearch_and_cassandra_mnesia / 7f966dd85b61080ccb6993f5b9023fa056a21379
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4260/214747/elasticsearch_and_cassandra_mnesia.26.2.3-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4260/214747/elasticsearch_and_cassandra_mnesia.26.2.3-amd64/big/ct_run.test%40934ee492c928.2024-04-10_12.45.40/index.html)
**OK: 437** / Failed: 0 / **User-skipped: 43** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/214804) / elasticsearch_and_cassandra_mnesia / e2d03e3c22a94b4841606850cd5f6819a8025449
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4260/214804/elasticsearch_and_cassandra_mnesia.26.2.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4260/214804/elasticsearch_and_cassandra_mnesia.26.2.2-amd64/big/ct_run.test%40363640b98b33.2024-04-10_14.24.24/index.html)
**OK: 437** / Failed: 0 / **User-skipped: 43** / Auto-skipped: 0

File: README.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
-[![Codecov](https://codecov.io/gh/esl/MongooseIM/branch/rel-6.1/graph/badge.svg)](https://app.codecov.io/gh/esl/MongooseIM/tree/rel-6.1)
-[![GitHub Actions](https://github.com/esl/MongooseIM/workflows/CI/badge.svg?branch=rel-6.1)](https://github.com/esl/MongooseIM/actions/workflows/ci.yml?query=branch%3Arel-6.1)
-[![Coveralls](https://coveralls.io/repos/github/esl/MongooseIM/badge.svg?branch=rel-6.1)](https://coveralls.io/github/esl/MongooseIM?branch=rel-6.1)
+[![CircleCI](https://dl.circleci.com/status-badge/img/gh/esl/MongooseIM/tree/rel-6.2.svg?style=shield)](https://app.circleci.com/pipelines/github/esl/MongooseIM?branch=rel-6.2)
+[![Codecov](https://codecov.io/gh/esl/MongooseIM/branch/rel-6.2/graph/badge.svg)](https://app.codecov.io/gh/esl/MongooseIM/tree/rel-6.2)
+[![GitHub Actions](https://github.com/esl/MongooseIM/workflows/CI/badge.svg?branch=rel-6.2)](https://github.com/esl/MongooseIM/actions/workflows/ci.yml?query=branch%3Arel-6.2)
+[![Coveralls](https://coveralls.io/repos/github/esl/MongooseIM/badge.svg?branch=rel-6.2)](https://coveralls.io/github/esl/MongooseIM?branch=rel-6.2)
```
Comment (NelsonVides) [inline README.md:7]: Why not just using the master branch in all of these? 🤔

Comment (chrzaszcz) [reply README.md:7]: Because it would fail very often. Let's just keep it in sync with docs, which already have `rel-6.2`.

**Review by jacekwegr — APPROVED** [review]
Looks good to me! 👍

**Pull request merged**